doc/user/unit-testing.md

```diff
@@ -1,132 +1,262 @@
 # Unit testing
+------------
 
-Testing your operator should involve both unit and [end-to-end][doc_e2e_test] tests. Unit tests assess the expected outcomes of individual operator components without requiring coordination between components. Operator unit tests should test multiple scenarios likely to be encountered by your custom operator logic at runtime. Much of your custom logic will involve API server calls via a [client][doc_client]; `Reconcile()` in particular will be making API calls on each reconciliation loop. These API calls can be mocked by using `controller-runtime`'s [fake client][doc_cr_fake_client], perfect for unit testing. This document steps through writing a unit test for the [memcached-operator][repo_memcached_reconcile]'s `Reconcile()` method using a fake client.
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  <!-- *generated with [DocToc](https://github.com/thlorenz/doctoc)*  -->
 
-## Fake client
+- [Overview](#overview)
+- [Using a Fake client](#using-a-fake-client)
+- [Testing Reconcile](#testing-reconcile)
+- [Testing with 3rd Party Resources](#testing-with-3rd-party-resources)
+- [How to increase the verbosity of the logs?](#how-to-increase-the-verbosity-of-the-logs)
 
-`controller-runtime`'s fake client exposes the same set of operations as a typical client, but simply tracks objects rather than sending requests over a network. You can create a new fake client that tracks an initial set of objects with the following code:
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-```Go
-import (
-  "testing"
+## Overview
+
+Testing your operator should involve both unit and [e2e][doc-e2e-test] tests. Unit tests assess the expected outcomes of individual operator components without requiring coordination between components. Operator unit tests should test multiple scenarios likely to be encountered by your custom operator logic at runtime. Much of your custom logic will involve API server calls via a [client][doc-client]; `Reconcile()` in particular will be making API calls on each reconciliation loop. These API calls can be mocked by using `controller-runtime`'s [fake client][doc-cr-fake-client], perfect for unit testing. This document steps through writing a unit test for the [memcached-operator][repo-memcached-reconcile]'s `Reconcile()` method using a fake client.
```

Comment:
I would revert back the change you made to e2e, "end-to-end" is clearer in my opinion as we are explaining in detail.

Reply:
Done.

Reply:
@AlexNPavel really sorry. I push this small change and lost yours over it. Could you please check it?

Reply:
FYI: If you do `--force-with-lease` you can prevent things like these from happening in the future :)

Also pinging @joelanford @hasbro17 to re-review ;)

Reply:
Just to share the `--force-with-lease` also lose the pr contributions :-(
But I applied the contribution in  the original one 👍